content/developer/howtos/upgrade_custom_db.rst

```diff
@@ -0,0 +1,305 @@
+
+  - :ref:`Make them installable <upgrade/empty_database/modules_installable>`
+  - :ref:`Test and fixes <upgrade/empty_database/test_fixes>`
+  - :ref:`Clean the code <upgrade/empty_database/clean_code>`
+  - :ref:`Make standard tests run successfully <upgrade/empty_database/standard_test>`
+
+.. TODO Check if needed
+.. At this stage, all the custom modules will be properly upgraded and fully functioning on an empty
+.. database and their code clean.
+.. It is ready to make the custom modules installable on the upgraded database.
+
+.. _upgrade/empty_database/modules_installable:
+
+Make custom modules installable
+-------------------------------
+
+The first step is to make the custom modules installable in the new Odoo version.
+This means, in a first instance, making sure there is no traceback or warnings when installing them.
+For this, install the custom modules (ideally, one by one) in an empty database targeting the new
+Odoo version and fix the tracebacks and warnings that arise from that.
+
+.. TODO Re-check and explain better the examples 
+
+This process will help detect issues during the installation of the modules.
+For example, invalid module dependencies, syntax changes (like the assets declaration, OWL updates,
+mail templates from Jinja to Qweb), non existing references to standard fields, models, views, etc.
+
+.. _upgrade/empty_database/test_fixes:
+
+Test and fixes
+--------------
+
+Once there are no more tracebacks when installing the modules, the next step is to test them.
+Even if the custom modules are installable in an empty database, this does not warranties there are
+no errors during their execution.
+Because of this, we encourage to test thoroughly all the customization to make sure everything
+is working as expected.
+
+This process will help detect further issues that are not identified during the module installation
+and can only be detected in runtime.
+For example, deprecated calls to standard python or OWL functions, non existing references to
+standard fields, etc.
+
+.. TODO Expand in the tests and their explanation
+
+We recommend to test all the customization, specially the following elements:
+
+  - Views
+  - Mail templates
+  - Reports
+  - Server actions and automated actions
+  - Changes in the standard workflows
+
+We also encourage to write automated tests to save time during the testing iterations, increase
+the coverage of the tests, and ensure that the changes and fixes introduced do not break the
+existing flows.
+If there are tests already implemented in the customization, make sure they are upgraded to the new
+Odoo version and run successfully, fixing issues that might be present.
+
+.. _upgrade/empty_database/clean_code:
+
+Clean the code
+--------------
+
+At this stage of the upgrade process, we also suggest to clean the code as much as possible.
+This includes: 
+
+  - Remove redundant and unnecessary code.
+  - Remove features that are now part of Odoo standard.
```

Comment:
This is already explained in section "stop the development

Reply:
Should we include a reference to that part of the doc or remove this point?

Reply:
Absolutely!